rustsec/src/advisory/metadata.rs

```diff
@@ -63,13 +63,17 @@ pub struct Metadata {
     /// affecting a particular crate without failing the build.
     pub informational: Option<Informational>,
 
-    /// URL with an announcement (e.g. blog post, PR, disclosure issue, CVE)
-    pub url: Option<Url>,
-
     /// Additional reference URLs with more information related to this advisory
     #[serde(default)]
     pub references: Vec<Url>,
 
+    /// URL with an announcement (e.g. blog post, PR, disclosure issue, CVE)
+    pub url: Option<Url>,
+
+    /// URL for the source registry where the vulnerable package is published
+    /// (defaults to crates.io, i.e. `registry+https://github.com/rust-lang/crates.io-index`)
+    pub registry: Option<Url>,
```

Comment:
One question: should this be a `Vec<Url>` (or perhaps `Option<Vec<Url>>`?)

I think `Option<Url>` is probably okay for now. Perhaps there will be a future where people are syndicating packages to multiple repositories, but this doesn't seem to be a particularly common thing with any existing packaging system.

Another alternative to consider is a `pub source: Option<SourceId>` which would permit sources other than package registries. I actually tried that originally and it seemed to overcomplicate things, and I couldn't really envision any use cases for filing advisories for anything other than packages published to registries.

Reply:
GHSA allows publishing advisories about Git repos, without requiring a package - but they don't include those in their curated set AFAIK. But if we support that, we also need a way to specify affected versions as git commit ranges - which OSV does, but I think will vastly overcomplicate things for us.

Reply:
There isn't currently, although it's easy enough to do today by running `cargo audit` multiple times with different `-d` flags

Reply:
Re: git, I could see the use case there (e.g. a company has an internal monorepo of crates sourced from git rather than a registry, and wants to have an internal advisory DB for it, possibly in the same repo)

Reply:
I've changed the field to `source` and made it store a full `SourceId`. This makes it possible to potentially support git-based sources in the future.

For now I suggest linting the field to ensure it's `registry+https://`